dozer/cogs/firstqa.py

```diff
@@ -100,6 +101,49 @@ async def frcqa(self, ctx: DozerContext, question: int):
     """
 
 
+    @commands.hybrid_command(name = "frcrule", pass_context = True)
+    @bot_has_permissions(embed_links = True)
+    @app_commands.describe(rule = "The rule number")
+    async def frcrule(self, ctx: DozerContext, rule: str):
+        letter_part = ''.join([char for char in rule if char.isalpha()])
+        number_part = ''.join([char for char in rule if char.isdigit()])
+        embed = discord.Embed(
+            title=f"Rule {letter_part.upper()}{number_part}",
+            url=f"https://frc-qa.firstinspires.org/manual/rule/{letter_part.upper()}/{number_part}",
+            color=discord.Color.blue()
+        )
+
+
+        if not re.match(r'^[a-zA-Z]\d{3}$', rule):
```

Comment:
If we're already using regex, I would much rather use regex groups to capture the letter and number part of the rule. Example using named groups: `^(?<letter>[a-zA-Z])(?<number>\d{3})$`